tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,771 @@
+				Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+				frrPod := deployTestPods(addressPool, hubIPaddresses, externalAdvertisedIPv4Routes,
+					externalAdvertisedIPv6Routes)
+
+				By("Creating BGP Peers")
+				createBGPPeerAndVerifyIfItsReady(ipv4metalLbIPList[0], "", 64500, false, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, "IPv4", removePrefixFromIPList(nodeAddrList), addressPool, 32)
+
+				By("Create first frrconfiguration that receieves a single route")
+				err := createFrrConfiguration("frrconfig-filtered-1", ipv4metalLbIPList[0], 64500,
+					[]string{externalAdvertisedIPv4Routes[0], externalAdvertisedIPv6Routes[0]}, false)
+				Expect(err).ToNot(HaveOccurred(), "Fail to create a first FRRconfiguration")
+
+				frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+					LabelSelector: "app=frr-k8s",
+				})
+				Expect(err).ToNot(HaveOccurred(), "Fail to find Frrk8 pod list")
+
+				By("Verify that the node FRR pods advertises two routes")
+				err = verifyExternalAdvertisedRoutes(frrPod, ipv4NodeAddrList, externalAdvertisedIPv4Routes)
+				Expect(err).ToNot(HaveOccurred(), "Fail to find advertised routes")
+
+				By("Validate BGP received only the first route")
+				Eventually(func() string {
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(SatisfyAll(
+					ContainSubstring(externalAdvertisedIPv4Routes[0]),      // First IP address
+					Not(ContainSubstring(externalAdvertisedIPv4Routes[1])), // Second IP address
+				), "Fail to find all expected received routes")
+
+				By("Create second frrconfiguration that receives a single route")
+				err = createFrrConfiguration("frrconfig-filtered-2", ipv4metalLbIPList[0], 64500,
+					[]string{externalAdvertisedIPv4Routes[1], externalAdvertisedIPv6Routes[1]}, false)
+				Expect(err).ToNot(HaveOccurred(), "Fail to create a second FRRconfiguration")
+
+				By("Validate BGP received only the first route")
+				Eventually(func() string {
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(SatisfyAll(
+					ContainSubstring(externalAdvertisedIPv4Routes[0]), // First IP address
+					ContainSubstring(externalAdvertisedIPv4Routes[1]), // Second IP address
+				), "Fail to find all expected received routes")
+			})
+
+		It("Verify that a FRR speaker rejects a contrasting FRRConfiguration merge",
```

Comment:
wdyt about keeping only the part about the conflicting configs? seems the metallb / peering part is redundant when all you want to check is that it handles conflicting configs properly

Reply:
I agree. Removed the peering.